templates/items/details/details-weapon.hbs

```diff
@@ -0,0 +1,78 @@
+<fieldset>
+    <legend>{{ localize "DND5E.ItemWeaponDetails" }}</legend>
+
+    {{!-- Weapon Type --}}
+    {{ formField fields.type.fields.value value=source.type.value label="DND5E.ItemWeaponType" localize=true
+                 choices=config.weaponTypes }}
+
+    {{!-- Weapon Base --}}
+    {{ formField fields.type.fields.baseItem value=source.type.baseItem label="DND5E.ItemWeaponBase" localize=true
+                 choices=baseItems }}
+
+    {{!-- Proficiency --}}
+    {{#unless isMountable}}
+    {{ formField fields.proficient value=source.proficient blank="DND5E.Automatic" localize=true
+                 choices=config.weaponAndArmorProficiencyLevels }}
+    {{/unless}}
+
+    {{!-- Weapon Properties --}}
+    {{ formField fields.properties options=properties.options label="DND5E.ItemWeaponProperties" localize=true
+                 input=inputs.createMultiCheckboxInput stacked=true classes="checkbox-grid" }}
+
+    {{!-- Range --}}
```

Comment:
We also have the new `system.range.reach` that we should show if the reach property is set.